src/metabase/lib/segment.cljc

```diff
@@ -1,18 +1,79 @@
 (ns metabase.lib.segment
-  "A Metric is a saved MBQL query stage snippet with `:filter`. Segments are always boolean expressions."
+  "A Segment is a saved MBQL query stage snippet with `:filter`. Segments are always boolean expressions."
   (:require
    [metabase.lib.metadata :as lib.metadata]
    [metabase.lib.metadata.calculation :as lib.metadata.calculation]
-   [metabase.shared.util.i18n :as i18n]))
+   [metabase.lib.metadata.protocols :as lib.metadata.protocols]
+   [metabase.lib.options :as lib.options]
+   [metabase.lib.ref :as lib.ref]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.schema.id :as lib.schema.id]
+   [metabase.lib.util :as lib.util]
+   [metabase.shared.util.i18n :as i18n]
+   [metabase.util.malli :as mu]))
+
+(defn- resolve-segment [query segment-id]
+  (when (integer? segment-id)
```

Comment:
Are named segments irrecoverable? I see this is the same as metrics.

Reply:
Yes, I kept to the metrics way.